livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/assistant_llm.py

```diff
@@ -0,0 +1,246 @@
+# Copyright 2023 LiveKit, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import asyncio
+import uuid
+from dataclasses import dataclass, field
+from typing import Any, MutableSet
+
+import httpx
+from livekit.agents import llm
+
+from openai import AsyncAssistantEventHandler, AsyncClient
+from openai.types.beta.threads import Text, TextDelta
+from openai.types.beta.threads.runs import ToolCall, ToolCallDelta
+
+from .models import AssistantTools, ChatModels
+from .utils import build_oai_message
+
+DEFAULT_MODEL = "gpt-4o"
+MESSAGE_ID_KEY = "__openai_message_id__"
+MESSAGES_ADDED_KEY = "__openai_messages_added__"
+
+
+@dataclass
+class LLMOptions:
+    model: str | ChatModels
+
+
+@dataclass
+class AssistantCreateOptions:
+    name: str
+    instructions: str
+    model: ChatModels
+    tools: list[AssistantTools] = field(default_factory=list)
+
+
+@dataclass
+class AssistantLoadOptions:
+    assistant_id: str
+    thread_id: str
+
+
+class AssistantLLM(llm.LLM):
+    def __init__(
+        self,
+        *,
+        assistant_opts: AssistantCreateOptions | AssistantLoadOptions,
+        client: AsyncClient | None = None,
+        api_key: str | None = None,
+        base_url: str | None = None,
+    ) -> None:
+        self._client = client or AsyncClient(
+            api_key=api_key,
+            base_url=base_url,
+            http_client=httpx.AsyncClient(
+                timeout=5.0,
+                follow_redirects=True,
+                limits=httpx.Limits(
+                    max_connections=1000,
+                    max_keepalive_connections=100,
+                    keepalive_expiry=120,
+                ),
+            ),
+        )
+        self._assistant_opts = assistant_opts
```

Comment:
nit: a bit confusing to set both types to the same var, and then to check the same var later.. not knowing what type it is. seems cleaner to keep them separate and only sync if it's a `AssistantCreateOptions`